odoo/addons/base/transpiler/sourcemap_generator.py

```diff
@@ -0,0 +1,168 @@
+from functools import lru_cache
+
+
+class SourceMapGenerator:
+    """
+    The SourceMapGenerator creates the sourcemap maps the asset bundle to the js files.
+
+    What is a sourcemap ? (https://developer.mozilla.org/en-US/docs/Tools/Debugger/How_to/Use_a_source_map)
+    In brief: a source map is what makes possible to debug your processed/compiled/minified code as your were debugging
+    the original, non-altered source code. It is a file that provides a mapping original <=> processed for the
+    browser to read.
+
+    This implementation of the SourceMapGenerator is a translation and adaptation of this implementation
+    in js https://github.com/mozilla/source-map. For performance purposes, we have removed all unnecessary
+    functions/steps for our use case. This simpler version does a line by line mapping, with the ability to
+    add offsets at the start and end of a file. (when we have to add comments on top a transpiled file by example).
+    """
+    def __init__(self, **kwargs):
+        self._file = kwargs.get("file")
+        self._sourceRoot = kwargs.get("sourceRoot")
+        self._sources = {}
+        self._mappings = []
+        self._sourcesContents = {}
+        self._version = 3
+        self.cache = {}
+
+    def _serialize_mappings(self):
+        """
+        A source map mapping is encoded with the base 64 VLQ format.
+        This function encodes the readable source to the format.
+
+        :return the encoded content
+        """
+        previousGeneratedLine = 1
+        previousOriginalLine = 0
+        previousSource = 0
+        encoded_column = base64vlq_encode(0)
+        result = ""
+        for mapping in self._mappings:
+            if (mapping["generatedLine"] != previousGeneratedLine):
```

Comment:
Remove parenthesis, they're not useful in python (same in the next two conditionals)